### PR TITLE
update example of AST for TokenTypes.PARAMETERS

### PR DESCRIPTION
**Pull request by Venkatesh-24**
Closes: #9551

Screenshot from 2021-11-01 23-43-51

```
**Test.java**
public class Test {
  public void sum(int num1, int... nums) {
    int[] num = nums;
    int sum = num1;
    for (int i : num) {
      sum = sum + i;
    }
  }
}

**AST**
java -jar checkstyle-9.0.1-all.jar -t Test.java
`--CLASS_DEF -> CLASS_DEF [4:0]
    |--MODIFIERS -> MODIFIERS [4:0]
    |   `--LITERAL_PUBLIC -> public [4:0]
    |--LITERAL_CLASS -> class [4:7]
    |--IDENT -> Test [4:13]
    `--OBJBLOCK -> OBJBLOCK [4:18]
        |--LCURLY -> { [4:18]
        |--METHOD_DEF -> METHOD_DEF [8:2]
        |   |--MODIFIERS -> MODIFIERS [8:2]
        |   |   `--LITERAL_PUBLIC -> public [8:2]
        |   |--TYPE -> TYPE [8:9]
        |   |   `--LITERAL_VOID -> void [8:9]
        |   |--IDENT -> sum [8:14]
        |   |--LPAREN -> ( [8:17]
        |   |--PARAMETERS -> PARAMETERS [8:18]
        |   |   |--PARAMETER_DEF -> PARAMETER_DEF [8:18]
        |   |   |   |--MODIFIERS -> MODIFIERS [8:18]
        |   |   |   |--TYPE -> TYPE [8:18]
        |   |   |   |   `--LITERAL_INT -> int [8:18]
        |   |   |   `--IDENT -> num1 [8:22]
        |   |   |--COMMA -> , [8:26]
        |   |   `--PARAMETER_DEF -> PARAMETER_DEF [8:28]
        |   |       |--MODIFIERS -> MODIFIERS [8:28]
        |   |       |--TYPE -> TYPE [8:28]
        |   |       |   `--LITERAL_INT -> int [8:28]
        |   |       |--ELLIPSIS -> ... [8:31]
        |   |       `--IDENT -> nums [8:35]
        |   |--RPAREN -> ) [8:39]
        |   `--SLIST -> { [8:41]
        |       |--VARIABLE_DEF -> VARIABLE_DEF [9:4]
        |       |   |--MODIFIERS -> MODIFIERS [9:4]
        |       |   |--TYPE -> TYPE [9:4]
        |       |   |   |--LITERAL_INT -> int [9:4]
        |       |   |   `--ARRAY_DECLARATOR -> [ [9:7]
        |       |   |       `--RBRACK -> ] [9:8]
        |       |   |--IDENT -> num [9:10]
        |       |   `--ASSIGN -> = [9:14]
        |       |       `--EXPR -> EXPR [9:16]
        |       |           `--IDENT -> nums [9:16]
        |       |--SEMI -> ; [9:20]
        |       |--VARIABLE_DEF -> VARIABLE_DEF [10:4]
        |       |   |--MODIFIERS -> MODIFIERS [10:4]
        |       |   |--TYPE -> TYPE [10:4]
        |       |   |   `--LITERAL_INT -> int [10:4]
        |       |   |--IDENT -> sum [10:8]
        |       |   `--ASSIGN -> = [10:12]
        |       |       `--EXPR -> EXPR [10:14]
        |       |           `--IDENT -> num1 [10:14]
        |       |--SEMI -> ; [10:18]
        |       |--LITERAL_FOR -> for [11:4]
        |       |   |--LPAREN -> ( [11:8]
        |       |   |--FOR_EACH_CLAUSE -> FOR_EACH_CLAUSE [11:9]
        |       |   |   |--VARIABLE_DEF -> VARIABLE_DEF [11:9]
        |       |   |   |   |--MODIFIERS -> MODIFIERS [11:9]
        |       |   |   |   |--TYPE -> TYPE [11:9]
        |       |   |   |   |   `--LITERAL_INT -> int [11:9]
        |       |   |   |   `--IDENT -> i [11:13]
        |       |   |   |--COLON -> : [11:15]
        |       |   |   `--EXPR -> EXPR [11:17]
        |       |   |       `--IDENT -> num [11:17]
        |       |   |--RPAREN -> ) [11:20]
        |       |   `--SLIST -> { [11:22]
        |       |       |--EXPR -> EXPR [12:10]
        |       |       |   `--ASSIGN -> = [12:10]
        |       |       |       |--IDENT -> sum [12:6]
        |       |       |       `--PLUS -> + [12:16]
        |       |       |           |--IDENT -> sum [12:12]
        |       |       |           `--IDENT -> i [12:18]
        |       |       |--SEMI -> ; [12:19]
        |       |       `--RCURLY -> } [13:4]
        |       `--RCURLY -> } [14:2]
        `--RCURLY -> } [15:0]

# Printing the code that we care
java -jar checkstyle-9.0.1-all.jar -t Test.java | grep -A9 "8:18"
        |   |--PARAMETERS -> PARAMETERS [8:18]
        |   |   |--PARAMETER_DEF -> PARAMETER_DEF [8:18]
        |   |   |   |--MODIFIERS -> MODIFIERS [8:18]
        |   |   |   |--TYPE -> TYPE [8:18]
        |   |   |   |   `--LITERAL_INT -> int [8:18]
        |   |   |   `--IDENT -> num1 [8:22]
        |   |   |--COMMA -> , [8:26]
        |   |   `--PARAMETER_DEF -> PARAMETER_DEF [8:28]
        |   |       |--MODIFIERS -> MODIFIERS [8:28]
        |   |       |--TYPE -> TYPE [8:28]
        |   |       |   `--LITERAL_INT -> int [8:28]
        |   |       |--ELLIPSIS -> ... [8:31]
        |   |       `--IDENT -> nums [8:35]
        |   |--RPAREN -> ) [8:39]
```
**Expected update for javadoc**

```
|--PARAMETERS -> PARAMETERS
        |   |   |--PARAMETER_DEF -> PARAMETER_DEF
        |   |   |   |--MODIFIERS -> MODIFIERS
        |   |   |   |--TYPE -> TYPE
        |   |   |   |   `--LITERAL_INT -> int
        |   |   |   `--IDENT -> num1
        |   |   |--COMMA -> ,
        |   |   `--PARAMETER_DEF -> PARAMETER_DEF
        |   |       |--MODIFIERS -> MODIFIERS
        |   |       |--TYPE -> TYPE
        |   |       |   `--LITERAL_INT -> int
        |   |       |--ELLIPSIS -> ... 
        |   |       `--IDENT -> nums 
        |   |--RPAREN -> )

```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -681,28 +681,20 @@
...
-     *     |   `--PARAMETER_DEF -&gt; PARAMETER_DEF
-     *     |       |--MODIFIERS -&gt; MODIFIERS
-     *     |       |--TYPE -&gt; TYPE
-     *     |       |   `--LITERAL_INT -&gt; int
-     *     |       |--ELLIPSIS -&gt; ...
-     *     |       `--IDENT -&gt; secondParameter
-     *     |--RPAREN -&gt; )
-     *      `--SLIST -&gt; {
-     *          `--RCURLY -&gt; }
+     * |--PARAMETERS -> PARAMETERS
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:684]: we need whole block of code to show AST in context of other nodes.
please keep METHOD_DEF

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (Venkatesh-24) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:684]: Okay sure I will keep the method definition also

Comment (Venkatesh-24) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:684]: Hi @romani I am getting merge conflicts while I was trying to pull and push my changed code
I think some other developer is working on it. Please tell me what am I supposed to do.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Please read attentively https://github.com/checkstyle/checkstyle/wiki/To-student-of-GSoC

Some links show how to rebase code over latest

**Pull request closed without merging**